pkg/sql/schemachanger/scplan/internal/scstage/build.go

```diff
@@ -462,7 +462,7 @@ func (sb stageBuilder) hasUnmeetableOutboundDeps(n *screl.Node) (ret bool) {
 		if !sb.isUnmetInboundDep(de) {
 			return nil
 		}
-		if de.Kind() != scgraph.SameStagePrecedence || sb.hasUnmeetableOutboundDeps(de.From()) {
+		if (de.Kind() != scgraph.SameStagePrecedence && de.Kind() != scgraph.Precedence) || sb.hasUnmeetableOutboundDeps(de.From()) {
```

Comment:
My understanding is that when we're considering scheduling a candidate node in a stage, we need to recursively visit the neighbouring nodes which would then have to also be scheduled in the same stage, in order to check that this would be possible in all cases. This same-stage constraint is exclusively imposed by the `scgraph.SameStagePrecedence` DepEdgeKind value.

I neither understand the bug right now, nor why this fix is correct. Can you provide an explanation/example/test/something to illustrate?